File: Arrays101/Introduction/MaxConsecutiveOnes.java
package Arrays101.Introduction;

public class MaxConsecutiveOnes {

    public static void main(String[] args) {
        int[] num = { 1, 1, 0, 1, 1, 1 };
        findMaxConsecutiveOnes(num);
    }

    public static int findMaxConsecutiveOnes(int[] nums) {
        int res = 0;
        int holder = 0;
        for (int i = 0; i < nums.length; i++) {
            if (nums[i] == 1) {
                holder++;
                if (holder > res) {
                    res = holder;
                }
            } else {
                holder = 0;
            }
        }
        return res;
    }
}

File: Arrays101/introduction/MaxConsecutiveOnes.py
def findMaxConsecutiveOnes(nums):
    if len(nums) == 0:
        return 0

    counter = 0
    ones = 0

    for i in range(len(nums) + 1):
        if nums[i] == 1:
            ones += 1
        else:
            if ones > counter:
                counter = ones
                ones = 0    
    return counter                    


num = [1,1,0,1,1,1]
findMaxConsecutiveOnes(num)        